src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
@shabanovd Okay so I like `startDocument` (and `endDocument`) rather than `startProcessing` (and `endProcessing`). As `startProcessing` implies to me a complete indexing run (e.g. multiple documents) as opposed to a single document, we might also like to signal reindexing operations by using something like `startProcessing` or `startIndexing`.

I also agree that the Indexer should ideally not keep state. However that was not a change that I was trying to make in this PR. Such a change would require a larger design of the indexing architecture to move the `pending` and `flush` operations into a stream based approach instead. So I won't change the mutable/immutable property of the IndexWorker just yet, maybe in future...